lib.php

```diff
@@ -0,0 +1,70 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Main library of plugin.
+ *
+ * @package    quizaccess_seb
+ * @author     Andrew Madden <andrewmadden@catalyst-au.net>
+ * @copyright  2019 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+defined('MOODLE_INTERNAL') || die();
+
+/**
+ * Serve a seb config file for a particular quiz.
+ *
+ * @param string $cmid The course module ID for a quiz with config.
+ * @return string SEB config string.
+ *
+ * @throws coding_exception
+ * @throws dml_exception
+ * @throws moodle_exception
+ * @throws require_login_exception
+ */
+function quizaccess_seb_get_config(string $cmid) : string {
+    // Try and get the course module.
+    $cm = get_coursemodule_from_id('quiz', $cmid, 0, false, MUST_EXIST);
+
+    // Make sure the user is logged in and has access to the module.
+    require_login($cm->course, false, $cm);
+
+    // Retrieve the config for quiz.
+    $settings = \quizaccess_seb\quiz_settings::get_record(['quizid' => $cm->instance]);
+    // If no settings found, config is false, otherwise get config.
+    $config = $settings !== false ? $settings->get('config') : false;
```

Comment:
I know that you can do this in PHP, but it looks complicated and it takes some time to understand. Wouldn't you mind to simplify?

Reply:
actually scratch that. There is comment